recipes-ni/initscripts-nilrt/files/nicreatelogdir

```diff
@@ -0,0 +1,11 @@
+#! /bin/sh
+# Copyright (c) 2012-2013 National Instruments.
+# All rights reserved.
+#
+# Ensure the log directory is always present with the correct
+# ownership.
+
+log_dir="/var/local/natinst/log"
+
+mkdir -p -m 777 $log_dir
+chown lvuser:ni $log_dir
```

Comment:
Do we really need an initscript for this? Surely there is either (a) already another initscript which creates NI-owned directories or (b) we can do what every other distro does and expect that log-creating packages will make their own directories.

Reply:
It does seem excessive to have an initscript just for this. But I [haven't been able to find](https://codesearch.natinst.com/search/text?q=mkdir.*natin%20repo%3Ameta-nilrt&fold_case=false&regex=true&context=true) another initscript that creates NI dirs.

As for (b), I suspect the reason this was added was to support some old piece of software that didn't creates its own dir. Do you think we should remove this to find all such cases and have their devs fix them?

Reply:
>Do you think we should remove this to find all such cases and have their devs fix them?

It [seems like](https://codesearch.natinst.com/search/text?q=local%2Fnatinst%2Flog&fold_case=auto&regex=true&context=true) that path is used in a lot of different packages. In all cases, it seems like the intention is just to give the `lvuser` user a location to access core dumps, LV crash logs, ect. (which implies that it is where LV customers are intended to store their logs).

It's probably too risky to just remove it from the base image and require the NIFeeds package owners to fix all their packages. But I might like to move this code to something like the [`niacctbase`](https://github.com/ni/meta-nilrt/blob/nilrt/master/dunfell/recipes-core/niacctbase/niacctbase.bb) package, which does generic setup for the `lvuser` account.

Reply:
I found a [recipe](https://github.com/ni/meta-nilrt/blob/nilrt/master/dunfell/recipes-support/syslog-ng/syslog-ng_3.%25.bbappend#L25) that creates the directory too.